src/processing/getFiatRate/utils.ts

```diff
@@ -1,15 +1,25 @@
 import { AxiosResponse } from 'axios';
 import { ITrade } from '../../types';
+import { getDayAvg } from './getDayAvgCurrencyRate';
 
-interface ICryptoCompareResponse {
-    [key: string]: number;
+export function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
 }
 
-export function cryptocompareRateResponse(response: AxiosResponse<ICryptoCompareResponse>, fiatCurrency: string) {
+export async function cryptocompareRateResponse(
+    response: AxiosResponse,
+    fiatCurrency: string,
+    currency: string,
```

Comment:
Im not I am understanding why we are adding these extra params to the call

Reply:
Because if it fails, we have to call getDayAvg again with the same parameters (ligne 21)

Reply:
Wouldnt it make more sense to simply add a check before line 22 here, https://github.com/starsoccer/cryptotithe/pull/276/files#diff-e3b86f754bffb6dd67bce4b0ebb16b8e723332be7d3c64649478b77ab09c08b9R22 confirming its not rate limited?

Reply:
True that

Reply:
What is the purpose of this cryptocompareRateResponse function at all? Can’t we put the whole thing in getDayAvg() itself, since it is only used there?

Reply:
I deleted the function all together and moved the processing back into getDayAvg()

Reply:
Fine with me. I think I moved it out as originally it was reused but no isnt anymore